reframe/frontend/cli.py

```diff
@@ -410,15 +411,27 @@ def main():
 
     # Setup the check loader
     if options.checkpath:
-        load_path = []
+        load_path = OrderedSet()
         for d in options.checkpath.split(':'):
             d = os_ext.expandvars(d)
             if not os.path.exists(d):
                 printer.warning("%s: path `%s' does not exist. Skipping..." %
                                 (argparser.prog, d))
                 continue
 
-            load_path.append(d)
+            d = os.path.abspath(os.path.realpath(d))
```

Comment:
Why do we need to get the `abspath` from `realpath`? why not take `abspath(d)`?

Reply:
In case it is a symbolic link.